core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -188,7 +191,10 @@ func (p *logEventProvider) GetLatestPayloads(context.Context) ([]ocr2keepers.Upk
 }
 
 // ReadLogs fetches the logs for the given upkeeps.
-func (p *logEventProvider) ReadLogs(ctx context.Context, force bool, ids ...*big.Int) error {
+func (p *logEventProvider) ReadLogs(pctx context.Context, force bool, ids ...*big.Int) error {
+	ctx, cancel := context.WithTimeout(pctx, readLogsTimeout)
+	defer cancel()
```

Comment:
not able to comment on the exact line, if this context cancels then we would get an error in line 204.
I see we then update filters in line 205, should the error handling (line 207) be done before that?

Reply:
There might be an error for some of the filters, therefore we always try to update last poll block of those that are have managed to read successfully